### PR TITLE
 KAFKA-8155: Add 2.1.1 release to system tests

### PR DESCRIPTION
**Pull request by mjsax**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/upgrade-system-tests-21/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java
Comment (mjsax) [inline streams/upgrade-system-tests-21/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java]: Java8 cleanup only

File: streams/upgrade-system-tests-21/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java
Comment (mjsax) [inline streams/upgrade-system-tests-21/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java]: Java8 cleanup only

**Comment by mjsax** [comment]
Triggered _all_ system tests: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2515/

Call for review @guozhangwang @bbejeck @vvcephei @ableegoldman @cadonna

**Comment by mjsax** [comment]
This should be cherry-picked to `2.2` branch. Not sure about `2.1` ?

File: streams/upgrade-system-tests-21/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java
Comment (mjsax) [reply streams/upgrade-system-tests-21/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java]: Similar cleanup https://github.com/apache/kafka/pull/6598

Comment (mjsax) [reply streams/upgrade-system-tests-21/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java]: Similar cleanup in https://github.com/apache/kafka/pull/6599

Comment (mjsax) [reply streams/upgrade-system-tests-21/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java]: Different PR for different modules to allow corresponding cherry-picking to older branches.

**Review by bbejeck — APPROVED** [review]

**Comment by bbejeck** [comment]
>This should be cherry-picked to 2.2 branch. Not sure about 2.1 ?

I'd say do `2.1` as well as IMHO we want the latest versions on the system tests.

**Comment by bbejeck** [comment]
system test failed, but it was during the setup, not an actual test failure.  So I've kicked off another build https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2518/

**Comment by mjsax** [comment]
Updated `Dockerfile`. Retrigger system tests: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2520/

**Comment by mjsax** [comment]
System test failed. It did not download `2.1.1` artifact -- not sure why -- maybe I started the build before it could sync on the docker image update. Retriggered:

https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2525/

**Comment by mjsax** [comment]
Rebased to resolve merge conflict. Also fixed the artifact problem. Retriggered system tests: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2590/

**Review by vvcephei — APPROVED** [review]
Thanks!

**Comment by guozhangwang** [comment]
LGTM.

**Comment by guozhangwang** [comment]
retest this please

**Comment by mjsax** [comment]
Currently system tests on `trunk` are broken -- we need to wait for a fix IMHO, before we can merge this. Thoughts?

**Comment by mjsax** [comment]
Rebased this to get latest system test fixed. Retriggered system tests: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2630/

**Comment by mjsax** [comment]
Java8 failed (test results are no longer available). Java11 passed.

Retest this please.

**Comment by mjsax** [comment]
System test results not available any longer. Retriggered: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2654/

**Comment by mjsax** [comment]
System tests passed.

Java8 failed (test results are no longer available). Java11 passed.

Retest this please.

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `2.3` and `2.2` branches.

**Comment by ijuma** [comment]
@mjsax you broke the 2.2 system tests due to a badly resolved conflict:

https://github.com/apache/kafka/commit/c450bfc2911fa121e18815a15b5fd7f087fd03ac#diff-9f5a3c3b9b36f4d6c32b238f5ec80b89R125

After the commit above, a single system test was being executed:

http://confluent-kafka-2-2-system-test-results.s3-us-west-2.amazonaws.com/2019-06-13--001.1560391755--apache--2.2--bda2a02/report.html

**Pull request merged**
